state/accountsDB.go

```diff
@@ -958,28 +959,33 @@ func (adb *AccountsDB) RootHash() ([]byte, error) {
 
 // RecreateTrie is used to reload the trie based on an existing rootHash
 func (adb *AccountsDB) RecreateTrie(rootHash []byte) error {
+	return adb.RecreateTrieFromEpoch(holders.NewRootHashHolder(rootHash, core.OptionalUint32{}))
+}
+
+// RecreateTrieFromEpoch is used to reload the trie based on the provided options
+func (adb *AccountsDB) RecreateTrieFromEpoch(options common.RootHashHolder) error {
 	adb.mutOp.Lock()
 	defer adb.mutOp.Unlock()
 
-	err := adb.recreateTrie(rootHash)
+	err := adb.recreateTrie(options)
 	if err != nil {
 		return err
 	}
-	adb.lastRootHash = rootHash
+	adb.lastRootHash = options.GetRootHash()
 
 	return nil
 }
 
-func (adb *AccountsDB) recreateTrie(rootHash []byte) error {
-	log.Trace("accountsDB.RecreateTrie", "root hash", rootHash)
+func (adb *AccountsDB) recreateTrie(options common.RootHashHolder) error {
+	log.Trace("accountsDB.RecreateTrie", "root hash", options.GetRootHash())
```

Comment:
for improved log prints: suggestion to add a String() method on the commonRootHashHolder and print directly the options instance here. the "root hash" log part can be changed to "root hash holder".

Reply:
Done.